docs/reconfigurator-dev-guide.adoc

```diff
@@ -0,0 +1,908 @@
+
+```
+$ TUF_REPO_DOWNLOAD_URL=https://buildomat.eng.oxide.computer/public/file/oxidecomputer/omicron/rot-all/$OMICRON_COMMIT/repo.zip
+```
+
+Now `cd` to the directory you want to download the TUF repo to.  You should have at least 4-5 GiB of free disk space (enough for the zipped and unzipped copies of the TUF repo).  We'll create a directory named for the commit:
+
+```
+$ mkdir $OMICRON_COMMIT
+$ cd $OMICRON_COMMIT
+```
+
+Download the repo with:
+
+```
+$ curl -L -C - -O $TUF_REPO_DOWNLOAD_URL
+```
+
+Sometimes this download gets interrupted.  If that happens, you can run the same command again to resume the download where it left off.
+
+For some of the workflows here, you'll want an _unpacked_ TUF repo.  You can unpack it with:
+
+```
+$ unzip FILENAME
+```
+
+This should create a directory called `repo` with subdirectories `metadata` and `targets`.
+
+
+[#task-trim-TUF-repo]
+=== Task: trim TUF repo for shipping to a switch zone
+
+On a4x2 or a racklette, it's handy to run `reconfigurator-sp-updater` and `repo-depot-standalone` from the switch zone, with the TUF repo you're using also in the switch zone.  But the switch zone generally doesn't have enough space for a full TUF repo.  You can work around this by deleting some large artifacts that we don't need for our purposes.
+
+Prerequisite: you must already have an _unpacked_ TUF repo.  You could <<task-download-TUF-repo,download one from CI>>.
+
+For testing SP updates, we don't need the host OS and control plane images, which are by far the largest files in the repo.  You can delete them with:
+
+```
+$ rm -f repo/targets/*.host* repo/targets/*.trampoline-* repo/targets/*.control-plane-*
+```
+
+Then copy this directory tree over to the switch zone.
+
+[#task-decide-sp-artifact]
+=== Task: decide which SP artifact you want to deploy
+
+Prerequisite:
+
+* You must have an unpacked TUF repo.
+
+You must first decide which SP you're going to update.  With simulated SPs (`cargo xtask omicron-dev run-all` and a4x2), this choice doesn't matter much.  With real hardware, it's a bigger deal because resetting the SP will reset the corresponding host.  You don't want to update the SP for the host you're doing your testing from!
+
+If you don't particularly care because you just to want to test update itself, sled 15 is a good choice on a racklette (since it's not a Scrimlet) and `SimGimlet00` (the first sled) is a good choice in simulated deployments.
+
+Once you've picked an SP, you need to know what kind of board it is.
+
+* With real hardware, it will be specific Gimlet revision (e.g., `gimlet-e`), Sidecar revision (`sidecar-c`), or PSC (e.g., `psc-c`).
+* With simulated SPs, it will be `SimGimletSp` or `SimSidecarSp`.
+
+Once you know which SP you're going to update, you can identify the board in one of two ways:
+
+* Using <<task-using-omdb,omdb>> to view inventory, you want the value of the BOARD column for the `SpSlot0` caboose.  (It will be the same for `SpSlot1`.)
```

Comment:
I think it's BORD right?

```suggestion
* Using <<task-using-omdb,omdb>> to view inventory, you want the value of the BORD column for the `SpSlot0` caboose.  (It will be the same for `SpSlot1`.)
```